### ISSUE TITLE
Better argument names for torch.atan2

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#61816]
Compare [`torch.atan2(input, other)`](https://pytorch.org/docs/stable/generated/torch.atan2.html) with [`float atan2 ( float y, float x )`](https://en.cppreference.com/w/cpp/numeric/math/atan2). I think, `y, x` is clearer and inherits arg traditions from old C functions

**Comment by Lezcano** [comment]
This makes sense. I think we'd accept a short PR fixing this. cc @mruberry

**Comment by heitorschueroff** [comment]
The [Python Array API Standard](https://data-apis.org/array-api/latest/API_specification/function_and_method_signatures.html) specifies that input tensors should be named `x1`, `x2`, ... or simply `x` for single input. Changing parameter names is BC breaking so we should consider the more general issue of inconsistent naming.

**Comment by rgommers** [comment]
> The [Python Array API Standard](https://data-apis.org/array-api/latest/API_specification/function_and_method_signatures.html) specifies that input tensors should be named `x1`, `x2`, ... or simply `x` for single input.

Not quite. It uses positional-only parameters exactly because of the different naming conventions. It's highly unusual anyway to write `arctan(some_tensor)` as `arctan(input=some_tensor)` or `arctan(x1=some_tensor)`, so better to forbid that so no library has to make any changes.

**Comment by rgommers** [comment]
`input` is a historically not-so-great choice because it's a Python keyword, but it really doesn't matter much. I'd suggest this is not worth the effort trying to change.

**Comment by heitorschueroff** [comment]
> > The [Python Array API Standard](https://data-apis.org/array-api/latest/API_specification/function_and_method_signatures.html) specifies that input tensors should be named `x1`, `x2`, ... or simply `x` for single input.
> 
> Not quite. It uses positional-only parameters exactly because of the different naming conventions. It's highly unusual anyway to write `arctan(some_tensor)` as `arctan(input=some_tensor)` or `arctan(x1=some_tensor)`, so better to forbid that so no library has to make any changes.

So what the documentation for the Array API wants to say is that, in the documentation, this is how the positional-only array parameters are named? If so, it should be made clearer that this is not a requirement for complying with the standard.

**Comment by Lezcano** [comment]
The notation is specified in point 1 of the URL you linked above, and it is also explicit via the `fun(x1, x2, /)` notation which was introduced in [Python 3.8](https://www.python.org/dev/peps/pep-0570/), but I do agree that it may be easy to miss.

If I remember correctly, @mruberry mentioned once that PyTorch does not necessarily try to be BC with respect to the names of the arguments of functions that do not have defaults. Now, @rgommers also has a point in that, so few people use these that fixing this may not even be of interest.

**Comment by vadimkantorov** [comment]
My argument is not about anyone actually trying to use these arguments as keyword arguments, but mainly about clarity of documentation, so that you don't have to think twice when encountering these strange names after working with more logical `y, x` (probably 60 years in use) or even `x1, x2`

**Comment by mruberry** [comment]
Yep, there are better names. I think this is only interesting to fix if we do a broader name review, however. Lots of our operators suffer from not great argument names.

(And yes, changing the name of positional arguments is technically BC-breaking, but like @rgommers says it's so rare that users specify positional arguments like by name that we typically don't bother with a deprecation.)

**Comment by vadimkantorov** [comment]
Agreed about broader name review

**Comment by vadimkantorov** [comment]
I think the main point in potentially changing the arg names (or introducing alias argnames) is to improve the readability of the docs (especially by new users)

**Comment by rgommers** [comment]
Aliases don't seem helpful, they just make things more messy. I think the thing to consider is to make these arguments positional-only. That is possible now that Python 3.8 is the minimum version. It's equally BC-breaking, but more generally useful. And then any later argument renames are simply stylistic cleanups that are guaranteed not to have any BC impact.

Scikit-learn went through a whole keyword-only change (see https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep009/proposal.html) and is quite happy with the results - there were few complaints. Positional-only I think would be less disruptive than keyword-only.

IIRC there is a separate issue about this, but I cannot find it right now.
